framework/topology.cpp

```diff
@@ -848,19 +849,14 @@ static Topology build_topology()
             core->threads.resize(info->thread_id + 1);
         Topology::Thread *thr = &core->threads[info->thread_id];
 
-        if (thr->cpu != -1) {
-            valid_topology = false;
-            break;
-        }
+        if (thr->cpu != -1)
+            return Topology({});
 
-        thr->cpu = i;
+        thr->cpu = info - cpu_info;
```

Comment:
I really see this as an inconsistency. You moved the cycle to be C++ iterator-based to use C pointer arithmetic?..

Reply:
Yes, though this is going away in a future commit (#348)